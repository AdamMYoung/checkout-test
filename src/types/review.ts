import { SchemaOf, object, string, number } from 'yup';

/**
 * Payload to create a review.
 */
export type CreateReviewPayload = {
    /**
     * Email address of the user.
     */
    emailAddress: string;

    /**
     * Name of the reviewer.
     */
    name: string;

    /**
     * Rating of the review. Only accepts whole numbers between 1 and 5.
     */
    rating: number;

    /**
     * Comment on the review.
     */
    comment: string;
};

/**
 * A single review entry.
 */
export type Review = {
    /**
     * Name of the reviewer.
     */
    name: string;

    /**
     * Rating of the review. Only accepts whole numbers between 1 and 5.
     */
    rating: number;

    /**
     * Comment on the review.
     */
    comment: string;

    /**
     * Date the review was created.
     */
    createdAt: string;
};

/**
 * Schema to validate the payload of the `reviews` POST API endpoint.
 */
export const createReviewPayloadSchema: SchemaOf<CreateReviewPayload> = object({
    emailAddress: string().email('Please enter a valid email address').required('Please enter an email address'),
    name: string().required('Please enter your name'),
    rating: number()
        .min(1, 'Rating must be between 1 and 5')
        .max(5, 'Rating must be between 1 and 5')
        .integer()
        .required('Please select a rating'),
    comment: string().required('Please enter a comment'),
});
