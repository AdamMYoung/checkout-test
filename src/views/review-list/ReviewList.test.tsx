import { screen, render } from '@testing-library/react';
import { axe } from 'jest-axe';
import { ReviewList } from '.';
import { Review } from '../../types';

const reviews: Review[] = [
    { name: 'John', rating: 5, comment: 'Great!', createdAt: new Date().toISOString() },
    { name: 'Joe', rating: 4, comment: 'Good!', createdAt: new Date().toISOString() },
    { name: 'Jane', rating: 3, comment: 'Okay.', createdAt: new Date().toISOString() },
    { name: 'Bill', rating: 2, comment: 'Meh.', createdAt: new Date().toISOString() },
    { name: 'Bob', rating: 4, comment: 'Good!', createdAt: new Date().toISOString() },
    { name: 'Simon', rating: 5, comment: 'Great!', createdAt: new Date().toISOString() },
];

describe('ReviewList', () => {
    it('passes aXe validation', async () => {
        const { container } = render(<ReviewList reviews={reviews} />);

        expect(await axe(container)).toHaveNoViolations();
    });

    it('matches snapshot', async () => {
        const { container } = render(<ReviewList reviews={reviews} />);

        expect(container).toMatchSnapshot();
    });

    it('renders the comment section', async () => {
        render(<ReviewList reviews={reviews} />);

        const comment1 = screen.getByText(/John/i);
        const comment2 = screen.getByText(/Joe/i);
        const comment3 = screen.getByText(/Jane/i);
        const comment4 = screen.getByText(/Bill/i);
        const comment5 = screen.getByText(/Bob/i);

        expect(comment1).toBeInTheDocument();
        expect(comment2).toBeInTheDocument();
        expect(comment3).toBeInTheDocument();
        expect(comment4).toBeInTheDocument();
        expect(comment5).toBeInTheDocument();
    });
});
