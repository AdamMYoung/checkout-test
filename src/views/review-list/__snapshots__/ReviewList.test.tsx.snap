// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewList matches snapshot 1`] = `
<div>
  <div
    class="chakra-stack css-j7qwjs"
  >
    <article
      class="chakra-stack css-n21gh5"
    >
      <div
        class="chakra-stack css-1h3nfwc"
      >
        <div
          class="chakra-stack css-1qwdsol"
        >
          <span
            aria-label="5 out of 5 stars"
            class="chakra-stack css-193rmy8"
            role="img"
          >
            <svg
              class="chakra-icon css-1k73fkr"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="chakra-icon css-1k73fkr"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="chakra-icon css-1k73fkr"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="chakra-icon css-1k73fkr"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="chakra-icon css-1k73fkr"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
          </span>
          <h3
            class="chakra-heading css-1tapsqs"
          >
            John
          </h3>
        </div>
        <p
          class="chakra-text css-5ft5pb"
        >
          31 January 2022
        </p>
      </div>
      <p
        class="chakra-text css-0"
      >
        Great!
      </p>
    </article>
    <hr
      aria-orientation="horizontal"
      class="chakra-divider css-1y67kn3"
    />
    <article
      class="chakra-stack css-n21gh5"
    >
      <div
        class="chakra-stack css-1h3nfwc"
      >
        <div
          class="chakra-stack css-1qwdsol"
        >
          <span
            aria-label="4 out of 5 stars"
            class="chakra-stack css-193rmy8"
            role="img"
          >
            <svg
              class="chakra-icon css-1k73fkr"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="chakra-icon css-1k73fkr"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="chakra-icon css-1k73fkr"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="chakra-icon css-1k73fkr"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="chakra-icon css-6aos0a"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
          </span>
          <h3
            class="chakra-heading css-1tapsqs"
          >
            Joe
          </h3>
        </div>
        <p
          class="chakra-text css-5ft5pb"
        >
          31 January 2022
        </p>
      </div>
      <p
        class="chakra-text css-0"
      >
        Good!
      </p>
    </article>
    <hr
      aria-orientation="horizontal"
      class="chakra-divider css-1y67kn3"
    />
    <article
      class="chakra-stack css-n21gh5"
    >
      <div
        class="chakra-stack css-1h3nfwc"
      >
        <div
          class="chakra-stack css-1qwdsol"
        >
          <span
            aria-label="3 out of 5 stars"
            class="chakra-stack css-193rmy8"
            role="img"
          >
            <svg
              class="chakra-icon css-1k73fkr"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="chakra-icon css-1k73fkr"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="chakra-icon css-1k73fkr"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="chakra-icon css-6aos0a"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="chakra-icon css-6aos0a"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
          </span>
          <h3
            class="chakra-heading css-1tapsqs"
          >
            Jane
          </h3>
        </div>
        <p
          class="chakra-text css-5ft5pb"
        >
          31 January 2022
        </p>
      </div>
      <p
        class="chakra-text css-0"
      >
        Okay.
      </p>
    </article>
    <hr
      aria-orientation="horizontal"
      class="chakra-divider css-1y67kn3"
    />
    <article
      class="chakra-stack css-n21gh5"
    >
      <div
        class="chakra-stack css-1h3nfwc"
      >
        <div
          class="chakra-stack css-1qwdsol"
        >
          <span
            aria-label="2 out of 5 stars"
            class="chakra-stack css-193rmy8"
            role="img"
          >
            <svg
              class="chakra-icon css-1k73fkr"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="chakra-icon css-1k73fkr"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="chakra-icon css-6aos0a"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="chakra-icon css-6aos0a"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="chakra-icon css-6aos0a"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
          </span>
          <h3
            class="chakra-heading css-1tapsqs"
          >
            Bill
          </h3>
        </div>
        <p
          class="chakra-text css-5ft5pb"
        >
          31 January 2022
        </p>
      </div>
      <p
        class="chakra-text css-0"
      >
        Meh.
      </p>
    </article>
    <hr
      aria-orientation="horizontal"
      class="chakra-divider css-1y67kn3"
    />
    <article
      class="chakra-stack css-n21gh5"
    >
      <div
        class="chakra-stack css-1h3nfwc"
      >
        <div
          class="chakra-stack css-1qwdsol"
        >
          <span
            aria-label="4 out of 5 stars"
            class="chakra-stack css-193rmy8"
            role="img"
          >
            <svg
              class="chakra-icon css-1k73fkr"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="chakra-icon css-1k73fkr"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="chakra-icon css-1k73fkr"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="chakra-icon css-1k73fkr"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="chakra-icon css-6aos0a"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
          </span>
          <h3
            class="chakra-heading css-1tapsqs"
          >
            Bob
          </h3>
        </div>
        <p
          class="chakra-text css-5ft5pb"
        >
          31 January 2022
        </p>
      </div>
      <p
        class="chakra-text css-0"
      >
        Good!
      </p>
    </article>
    <hr
      aria-orientation="horizontal"
      class="chakra-divider css-1y67kn3"
    />
    <article
      class="chakra-stack css-n21gh5"
    >
      <div
        class="chakra-stack css-1h3nfwc"
      >
        <div
          class="chakra-stack css-1qwdsol"
        >
          <span
            aria-label="5 out of 5 stars"
            class="chakra-stack css-193rmy8"
            role="img"
          >
            <svg
              class="chakra-icon css-1k73fkr"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="chakra-icon css-1k73fkr"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="chakra-icon css-1k73fkr"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="chakra-icon css-1k73fkr"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
            <svg
              class="chakra-icon css-1k73fkr"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M23.555,8.729a1.505,1.505,0,0,0-1.406-.98H16.062a.5.5,0,0,1-.472-.334L13.405,1.222a1.5,1.5,0,0,0-2.81,0l-.005.016L8.41,7.415a.5.5,0,0,1-.471.334H1.85A1.5,1.5,0,0,0,.887,10.4l5.184,4.3a.5.5,0,0,1,.155.543L4.048,21.774a1.5,1.5,0,0,0,2.31,1.684l5.346-3.92a.5.5,0,0,1,.591,0l5.344,3.919a1.5,1.5,0,0,0,2.312-1.683l-2.178-6.535a.5.5,0,0,1,.155-.543l5.194-4.306A1.5,1.5,0,0,0,23.555,8.729Z"
                fill="currentColor"
              />
            </svg>
          </span>
          <h3
            class="chakra-heading css-1tapsqs"
          >
            Simon
          </h3>
        </div>
        <p
          class="chakra-text css-5ft5pb"
        >
          31 January 2022
        </p>
      </div>
      <p
        class="chakra-text css-0"
      >
        Great!
      </p>
    </article>
  </div>
</div>
`;
